src/compiler/moduleSpecifiers.ts

```diff
@@ -832,17 +847,54 @@ const enum MatchingMode {
     Pattern,
 }
 
-function tryGetModuleNameFromExports(options: CompilerOptions, targetFilePath: string, packageDirectory: string, packageName: string, exports: unknown, conditions: string[], mode = MatchingMode.Exact): { moduleFileToTry: string; } | undefined {
+function getOutputPath(targetFilePath: string, options: CompilerOptions, commonDir: string) {

```

Comment:
This is based on helpers from `src/compiler/emitter.ts`. I couldn't straight up reuse them though because I use different parameters list, the type of `options` is different etc.